### PR TITLE
Add TypeScript declaration file (index.d.ts)

### PR DESCRIPTION
**Pull request by lucasrmendonca**
This solves #12

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export = launchWebAuthFlow;
+
+declare function launchWebAuthFlow(options: WebAuthFlowOptions): ResponseUrl;
```
Comment (eight04) [inline index.d.ts]: Is there a way to tell typescript that this function returns a promise? i.e. this is an async function.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+interface WebAuthFlowOptions {
+  url: string,
+  redirect_uri: string,
+  interactive?: boolean,
+  alwaysUseTab?: boolean,
+  windowOptions?: {
```
Comment (eight04) [inline index.d.ts]: Is it possible to reuse type from other sources? Is there an official/3rd-party webextensions API type declaration that can be used?

**Comment by eight04** [comment]
Is there a linter that works with `.d.ts` files similar to eslint? Since this file is written manually by hand, it may include errors.

File: index.d.ts (hunk 2 above)
Comment (lucasrmendonca) [reply index.d.ts]: Mozilla has an official auto-generated type declaration https://www.npmjs.com/package/@types/webextension-polyfill
however, I'm not sure how to create a type based on another type declaration, I'll investigate.

File: index.d.ts (hunk 1 above)
Comment (lucasrmendonca) [reply index.d.ts]: This can be done with `Promise<ResponseUrl>` I'll fix it in the next patch

**Comment by lucasrmendonca** [comment]
> Is there a linter that works with `.d.ts` files similar to eslint? Since this file is written manually by hand, it may include errors.

There's [TSLint](https://palantir.github.io/tslint/) and we can add unit tests too, for extra safety.

**Comment by lucasrmendonca** [comment]
New additions:
- Removed the "ResponseUrl" type since it was just an alias for type "string" anyways.
- Fix the return value from "string" to "Promise<string>"
- Re-use common types from "webextension-polyfill" (although we're importing the original module, TypeScript will fetch the official types from https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/webextension-polyfill)
- Installed some TypeScript ESLint plugins and ran "npm run test" and fixed all linter suggestions (also modified the npm script to work for both *.js and *.ts files);

File: package.json [hunk 3]
```diff
@@ -15,15 +15,20 @@
     "@babel/core": "^7.11.1",
     "@babel/preset-env": "^7.11.0",
     "@rollup/plugin-babel": "^5.2.0",
+    "@types/webextension-polyfill": "^0.9.1",
```
Comment (eight04) [inline package.json:18]: Should we move this into prod-dependency? What is the common practice for this kind of peer-dependency type?

Or we should just document it in readme?

Comment (eight04) [reply package.json:18]: Or add a `"peerDependencies": {"types/webextension-polyfill": ">=0.9"}`?

Comment (lucasrmendonca) [reply package.json:18]: hmm, it seems that the standard practice for TypeScript libraries is to move all imported `@types/**` dependencies into prod-dependency... however, I'm against bloating the dist for something that won't be used for all users, only those who are using TS.
What if we kept it as a devDependency and just documented it to the readme, like this?

Installing via npm:
`npm install webext-launch-web-auth-flow`

Installing via npm with TypeScript support:
`npm install webext-launch-web-auth-flow @types/webextension-polyfill`

?

**Pull request merged**